src/lib/types/stores/account-store.ts

```diff
@@ -0,0 +1,33 @@
+import { IUser } from '../user';
+import { Store } from './store';
+
+export interface ICreateUser {
+    name?: string;
+    username?: string;
+    email?: string;
+    imageUrl?: string;
+}
+
+export interface IUserLookup {
+    id?: number;
+    username?: string;
+    email?: string;
+}
+
+export interface IUserUpdateFields {
+    name?: string;
+    email?: string;
+}
+
+export interface IAccountStore extends Store<IUser, number> {
+    update(id: number, fields: IUserUpdateFields): Promise<IUser>;
+    insert(user: ICreateUser): Promise<IUser>;
+    upsert(user: ICreateUser): Promise<IUser>;
```

Comment:
If we are adding these methods here, I wonder if we should remove the ones from user-store (or proxy them to this method), but because we're not calling them (at least not in this PR apparently), maybe these two can be removed from here.

Reply:
There are very good points, and if we ever want to create a new `user` or `service-account` we should use their respective stores, not `account-store`. Addressed in: https://github.com/Unleash/unleash/pull/2918/commits/bd671eb83267b91274d027cbe615d261de59800d